### ISSUE TITLE
 When hovering when the tooltip of stacked bar graph is set to shared, another bar graph tooltip is displayed

### ISSUE THREAD

**Issue by activeguild** [apexcharts/apexcharts.js#1639]
# Bug report

## use version
`apexcharts`: 3.19.2
`vue-apexcharts`: 1.5.2

## Codesandbox
https://codesandbox.io/s/vue-basic-example-40bqr

## Explanation

- If you hover over a bar graph when the following conditions are met, the next bar graph balloon will be displayed.
  - `chartOptions.chart. type `: 'bar'
  - `chartOptions.chart.stacked`: true
  - `chartOptions.tooltip. shared `: true

```js
      chartOptions: {
        chart: {
          type: "bar",
          stacked: true,
        },
        tooltip: {
          shared: true,
        },
        xaxis: {
          type: "datetime",
        },
      }
```

**Comment by elitedesignsoftware** [comment]
I have found this error too.  The tool tip is based on the X axis but not relative to the actual column that you're hovering on.  
May 3, 2021.

**Comment by lindapaiste** [comment]
I've been experiencing this issue myself and I've narrowed down which set of options cause the problem to appear.  In [this reproducible example](https://codesandbox.io/s/apexcharts-misaligned-hover-bl4c4?file=/src/index.js) the problematic combination is:

- `chart.stacked = true`
- `chart.type = "bar"`
- `xaxis.type = "datetime"`
- `tooltip.intersect = false`

Changing any one of those options will make the hover area line up properly.

If I change `chart.stacked` to `false` it's fine.
If I change `chart.type` to `"line"` it's fine.
If I change `xaxis.type` to `"category"` it's fine.
If I change `tooltip.intersect` to `true` then the hover position is fine -- but this means that `tooltip.shared` can no longer be used.

In other words, it's not possible to have a shared tooltip on a stacked datetime bar chart without a misaligned hover area.

**Comment by sptaszek92** [comment]
Is there any other solution for that issue? I need a multiple series chart with shared tooltip... but the hover area is really bad in that scenario...

**Comment by greengiraffe** [comment]
This issue is still present in version 3.35.0.

**Comment by Lukerayn3r** [comment]
Has this been resolved yet? I am using v3.49.1 and am still facing this exact problem.

**Comment by Lukerayn3r** [comment]
If anyone else comes across this issue I no longer share the tooltip
```
tooltip: {
      shared: false,
},
 ```
 
 And instead I create a custom tooltip which contains the shared data
 
 ```
 tooltip: {
    custom: function({series, seriesIndex, dataPointIndex, w}) {
        let xLabel = w.globals?.seriesX[0]?.[dataPointIndex] ?? w.globals.labels[dataPointIndex];
        let title = Number.isInteger(xLabel) ? moment(xLabel).format('ddd Do MMMM YYYY') : xLabel;
        let seriesLabels = '';

        let i = 0;
        for(const s of series) {
            // Skip iteration if series[i] is empty
            if (series[i][dataPointIndex] === undefined) {
                i++;
                continue;
            }

            seriesLabels += `
                <div><span class="series-color" style="background-color: `+ w.globals.colors[i] +`;"></span>` + w.globals.seriesNames[i].toLocaleString() + `: <b>` + series[i][dataPointIndex].toLocaleString() + `</b></div>
            `;
            i++;
        }

        return `<div class="tool-tip">
                    <div class="tool-tip-header text-capitalize"><b>`+ title +`</b></div>
                    <div class="tool-tip-body text-capitalize">
                        ` + seriesLabels + `
                        <div style="margin-top: 5px;">Total: <b>`+ w.globals.stackedSeriesTotals[dataPointIndex].toLocaleString() +`</b></div>
                    </div>
                </div>`
    },
    intersect: false
}
```
